sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -398,7 +398,7 @@ private[spark] object HiveUtils extends Logging {
         case childFirst: ChildFirstURLClassLoader =>
           childFirst.getURLs() ++ allJars(Utils.getSparkClassLoader)
         case urlClassLoader: URLClassLoader =>
-          urlClassLoader.getURLs ++ allJars(urlClassLoader.getParent)
+          allJars(urlClassLoader.getParent) ++ urlClassLoader.getURLs
```

Comment:
I am 100% sure this change will 'guarantee' the class loading order. Because in `ChildFirstURLClassLoader` we re-constructor it by setting the parent classloader to null and load classes in an order we explicitly set.